pkg/platform/runtime/runtime.go

```diff
@@ -47,12 +50,21 @@ func IsNeedsUpdateError(err error) bool {
 	return errs.Matches(err, &NeedsUpdateError{})
 }
 
-func newRuntime(target setup.Targeter, an analytics.Dispatcher, svcModel *model.SvcModel) (*Runtime, error) {
+// NeedsStageError is an error returned when the local runtime's build script has changes that need
+// staging. This is not a fatal error. A runtime can still be used, but a warning should be emitted.
+type NeedsStageError struct{ error }
+
+func IsNeedsStageError(err error) bool {
+	return errs.Matches(err, &NeedsStageError{})
+}
+
+func newRuntime(target setup.Targeter, an analytics.Dispatcher, svcModel *model.SvcModel, auth *authentication.Auth) (*Runtime, error) {
 	rt := &Runtime{
 		target:    target,
 		store:     store.New(target.Dir()),
 		analytics: an,
 		svcm:      svcModel,
+		auth:      auth,
 	}
 
 	if !rt.store.MarkerIsValid(target.CommitUUID()) {
```

Comment:
I think we should probably make this a single function, eg. `rt.CacheIsValid()`, which then in turn calls out to `rt.store.MarkerIsValid()` and the new logic for the buildscript. That way this common behavior is organized as such.